src/applications/vaos/services/appointment/index.js

```diff
@@ -119,47 +119,66 @@ function hasPartialResults(response) {
 export async function getBookedAppointments({
   startDate,
   endDate,
-  useV2 = false,
+  useV2VA = false,
+  useV2CC = false,
 }) {
   try {
-    if (useV2) {
-      const appointments = await getAppointments(startDate, endDate, [
-        'booked',
-        'cancelled',
-      ]);
-
-      const appointmentsWithoutRequests = appointments.filter(
-        appt => !appt.requestedPeriods,
+    let appointments = [];
+    if (!useV2VA) {
+      const confirmedVAAppointments = await getConfirmedAppointments(
+        'va',
+        moment(startDate).toISOString(),
+        moment(endDate).toISOString(),
       );
 
-      return transformVAOSAppointments(appointmentsWithoutRequests);
+      // We might get partial results back from MAS, so throw an error if we do
+      if (hasPartialResults(confirmedVAAppointments)) {
+        throw mapToFHIRErrors(
+          appointments[0].errors,
+          'MAS returned partial results',
+        );
+      }
+
+      appointments = [
+        ...transformConfirmedAppointments(confirmedVAAppointments.data),
+      ];
     }
 
-    const appointments = await Promise.all([
-      getConfirmedAppointments(
-        'va',
-        moment(startDate).toISOString(),
-        moment(endDate).toISOString(),
-      ),
-      getConfirmedAppointments(
+    if (!useV2CC) {
+      const confirmedCCAppointments = await getConfirmedAppointments(
```

Comment:
The awaits here change this function from making the two calls in parallel to making them serially. It would be nice to preserve the parallel fetching if we can without the code looking too confusing.